### PR TITLE
fix: back and forward does not work correctly, items should not have empty indexes

### PR DESCRIPTION
**Pull request by murataka**
In findindexes function , there's no need to

`item?.`

as the indexes are now being managed properly , I left them just because further testing might be essential and if there's still a mistake in the commit , not to make it crash.

It is possibly safe not to 

`item?.`.

### CODE REVIEW

**Comment by murataka** [comment]
Hello, 
let me know if you think this might create any other issue , and any kind of reviews would help. 

I would like to fix any possible issue created by this commit , as much as I can.

Thanks !

**Comment by murataka** [comment]
These comments were already explaining the below change .

```
       // There are two scenarios for creating an array with only one history record:
        // - When loaded id not found in the items array, this function by default will replace
        //   the first item. We need to keep only the new updated object, otherwise it will break
        //   the page when navigating forward in history.
        // - This is the first time any state modifications are done
        //   So we need to push the entry as there's nothing to replace
```
`items[index] = { path, state, id };` 
to
`items[0] = { path, state, id };`

File: packages/native/src/useLinking.tsx [hunk 1]
```diff
@@ -50,7 +50,7 @@ const createMemoryHistory = () => {
       const id = window.history.state?.id;
 
       if (id) {
-        const index = items.findIndex((item) => item.id === id);
+        const index = items.findIndex((item) => item && item.id === id);
```
Comment (roryabraham) [inline packages/native/src/useLinking.tsx:53]:
```suggestion
        const index = items.findIndex((item) => item?.id === id);
```

File: packages/native/src/useLinking.tsx [hunk 2]
```diff
@@ -99,7 +99,7 @@ const createMemoryHistory = () => {
 
       const id = window.history.state?.id ?? nanoid();
 
-      if (!items.length || items.findIndex((item) => item.id === id) < 0) {
+      if (!items.length || items.findIndex((item) => item && item.id === id) < 0) {
```
Comment (roryabraham) [inline packages/native/src/useLinking.tsx:102]:
```suggestion
      if (!items.length || items.findIndex((item) => item?.id === id) < 0) {
```

File: packages/native/src/useLinking.tsx [hunk 3]
```diff
@@ -179,7 +179,7 @@ const createMemoryHistory = () => {
 
         const onPopState = () => {
           const id = window.history.state?.id;
-          const currentIndex = items.findIndex((item) => item.id === id);
+          const currentIndex = items.findIndex((item) => item && item.id === id);
```
Comment (roryabraham) [inline packages/native/src/useLinking.tsx:182]:
```suggestion
          const currentIndex = items.findIndex((item) => item?.id === id);
```

**Review by roryabraham — APPROVED** [review]

**Review by satya164 — CHANGES_REQUESTED** [review]
This change doesn't make sense to me. The change seems to be that it's always going to replace the first item when calling replace regardless of what the index is which doesn't look correct and going to produce incorrect history.

**Comment by murataka** [comment]
> This change doesn't make sense to me. The change seems to be that it's always going to replace the first item when calling replace regardless of what the index is which doesn't look correct and going to produce incorrect history.

i shall check if the other solution if still creates empty indexes. feels still there's something missing..

**Comment by murataka** [comment]
https://github.com/react-navigation/react-navigation/pull/10312

solves the issue , and I did not get any empty indexes while testing it . 

Closing this pull request.

**Pull request closed without merging**